experiment_tool/dag.py

```diff
@@ -32,36 +32,34 @@ def _add_vertices(self, vertices):
         Add list of vertices to dag.
         """
         for v in vertices:
-            self.graph.setdefault(v, [])
+            self._graph.setdefault(v, [])
 
     def _add_edges(self, edges):
         """
         Add list of edges to dag.
         Throws exception if there is an end of an edge that is not in dag.
         """
         for (x, y) in edges:
-            if (x not in self.graph.keys()) or (y not in self.graph.keys()):
-                raise KeyError("No such vertex in a graph")
-            self.graph[x].append(y)
+            if x not in self._graph.keys() or y not in self._graph.keys():
```

Comment:
Remove `keys()` method, please. I wouldn't remove the brackets to make the code easier to read, but whatever you want.